### ISSUE TITLE
Support for plugin repository tokens

### ISSUE THREAD

**Issue by m-kuhn** [opengisch/qgis-plugin-ci#285]
plugins.qgis.org now supports tokens, see https://github.com/qgis/QGIS-Django/pull/326
It would be great to have support for this in here to use scoped tokens rather than personal access credentials for uploading plugins

**Comment by Guts** [comment]
Hi @Xpirix,

Can you share us some documentation (md, OpenAPI JSON/YAML...) where we can find details about the POST request required to work those API token please?

**Comment by Xpirix** [comment]
Hi @Guts 

Please find a POST request template at https://github.com/qgis/QGIS-Django/blob/master/readme.md#token-based-authentication. Please let me know if you need more specific details.

**Comment by Guts** [comment]
Thanks @Xpirix!

I was thinking about something like swagger but I think we can move on for now with this example.

Is there a development or test instance of plugins.qgis.org on which to base our unit tests in CI/CD to check if publishing works as expected?

**Comment by Xpirix** [comment]
Yes, swagger is a good idea. I will try to implement it along with the incoming resources hub API.

Please find the unit test for this feature at: https://github.com/qgis/QGIS-Django/blob/master/qgis-app/plugins/tests/test_token_auth.py

**Comment by Guts** [comment]
> Yes, swagger is a good idea. I will try to implement it along with the incoming resources hub API.

It would be really appreciated! Do you want to me creating an issue?

**Comment by Xpirix** [comment]
Yes please, thank you.

**Comment by Guts** [comment]
> Yes please, thank you.

Done in https://github.com/qgis/QGIS-Django/issues/473

**Comment by 3nids** [comment]
@Gustry You forgot to pass the headers in the requests :)

**Comment by 3nids** [comment]
also, files should be passed with `files` argument (and not `data`) https://www.w3schools.com/python/ref_requests_post.asp

**Comment by Xpirix** [comment]
That's correct @3nids. Good catch. Here is an example of a working script @Gustry : 

```python
import requests

from pathlib import Path

TOKEN = "XXX"
PACKAGE_NAME = "XXX"
ZIP_NAME = "XXX.zip"

server_url = f"https://plugins.qgis.org/plugins/api/{PACKAGE_NAME}/version/add/"


headers = {
    "Authorization": f"Bearer {TOKEN}"
}

zip_file = Path(__file__).parent / ZIP_NAME
print(zip_file.exists())

with open(zip_file, 'rb') as file:
    response = requests.post(server_url, files={'package': file}, headers=headers)

print(response.text)
print(response.status_code)
```
